### PR TITLE
Kedro Viz Static Website Hosting on GCP

### LINKED ISSUE #1675: Shareable Viz with GCP
## Description
This ticket is related to parent ticket https://github.com/kedro-org/kedro-viz/issues/1116.
This will enable users to host, publish and share their Kedro-Viz pipeline visualisation with a URL on  GCP object storage.

## Context
We previously enabled users to publish and share versions of their Kedro-Viz pipeline visualisation with a URL on AWS S3 in https://github.com/kedro-org/kedro-viz/issues/1116.

Enabling this feature will increase interoperability with the major cloud providers, providing Kedro-Viz access to non-technical stakeholders, and ultimately increasing adoption of Kedro-Viz. 

## Acceptance Criteria
A user is able to easily host and share a version of their Kedro-Viz pipeline visualisation with a URL via GCP by

- Configuring their storage account and region.
- Creating an access key.
- In the CLI, adding their environment variable with the access key.
- Running the kedro viz deploy command.

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #1675 

## Development notes

* Added GCPDeployer class to support GCP deployment for shareableviz
* Added pytests to cover GCP Deployment

**NOTE: This will be merged into the branch `feature/shareableviz-extended-support` after https://github.com/kedro-org/kedro-viz/pull/1708**

## QA notes

**Setup Environment for GCP**

Authorizing via Service Account Key  

**On GCP Console**
1. Create a project in GCP console
2. Create Service Account
    - Go to "IAM & Admin" > "Service Accounts."
    - Click "Create Service Account."
    - Provide a name and optional description.
3. Grant Roles
    - Select the service account.
    - Go to "Roles and Permissions."
    - Assign "Storage Object Creator" and "Storage Object User" role. You can explore more roles [here](https://cloud.google.com/iam/docs/understanding-roles)
4. Create Service Key
    - Go to the "Keys" tab.
    - Click "Add Key" and choose "JSON" format.
    - Download and securely store the key file.

**On command line before running kedro viz deploy**
Set environment variable `GOOGLE_APPLICATION_CREDENTIALS` to the downloaded key file's absolute path.

**Run deploy command**
`kedro viz deploy --platform=gcp endpoint=test-endpoint --bucket-name=shareableviz`

**Note:**
- Endpoint - This can be your load balancer ip address or if SSL is setup, it will be your domain pointing to the load balancer. Follow [this](https://cloud.google.com/storage/docs/hosting-static-website) for more details

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

**Comment by noklam** [comment]
Is this ready to be reviewed?

**Comment by ravi-kumar-pilla** [comment]
> Is this ready to be reviewed?

I created this PR from the azure branch. The work is done but there are some common files from https://github.com/kedro-org/kedro-viz/pull/1708. I would like to get this reviewed once #1708 is merged.

File: package/kedro_viz/integrations/deployment/gcp_deployer.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            file_list = glob.glob(f"{str(html_dir)}/**/*", recursive=True)
+
+            for local_file_path in file_list:
+                content_type, _ = mimetypes.guess_type(local_file_path)
+
+                # ignore directories
+                if content_type is None:  # pragma: no cover
+                    continue
+
+                relative_path = local_file_path[len(str(html_dir)) + 1 :]
```
Comment (noklam) [inline package/kedro_viz/integrations/deployment/gcp_deployer.py:52]: What does this mean?

File: package/kedro_viz/integrations/deployment/base_deployer.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def _write_heap_injected_index(self, html_content):
             with open(temp_file_path, "w", encoding="utf-8") as temp_index_file:
                 temp_index_file.write(html_content)
 
-            self._fs.put(temp_file_path, f"{self._path}/")
+            self._fs.put(temp_file_path, f"{self._path}/", content_type="text/html")
```
Comment (noklam) [inline package/kedro_viz/integrations/deployment/base_deployer.py:70]: Does it works for all deployers?

File: package/kedro_viz/integrations/deployment/gcp_deployer.py (hunk 1 above)
Comment (ravi-kumar-pilla) [reply package/kedro_viz/integrations/deployment/gcp_deployer.py:52]: This gives the relative path of the files to build the remote path. i.e.,

local_file_path -> Path(__file__).parent.parent.parent.absolute()/html/static/js/main.9a9b73c9.chunk.js
relative_path -> static/js/main.9a9b73c9.chunk.js
self._path -> {GCP_PROTOCOL}://{bucket-name}
remote path -> {self._path}/{relative_path}

File: package/kedro_viz/integrations/deployment/base_deployer.py (hunk 2 above)
Comment (ravi-kumar-pilla) [reply package/kedro_viz/integrations/deployment/base_deployer.py:70]: yes it works for AWS, GCP and local. AzureDeployer has its own `_write_heap_injected_index` method

**Review by noklam — APPROVED** [review]

**Pull request merged**